dozer-ingestion/aerospike/src/connector.rs

```diff
@@ -401,11 +409,9 @@ async fn map_event(
         return Err(AerospikeConnectorError::InvalidKeyValue(key.clone()));
     }
 
-    let set_name = match key.clone().get(1) {
-        None => Err(AerospikeConnectorError::NoSetNameFindInKey(key.clone())),
-        Some(Some(set_name)) => Ok(set_name.clone()),
-        Some(None) => Err(AerospikeConnectorError::SetNameIsNone(key.clone())),
-    }?;
+    let [_, Some(ref set_name), _, Some(ref pk_in_key)] = key.clone()[..] else {
```

Comment:
This checks the length, so you can skip checking it above. The key is owned, so let's avoid cloning it both here and at the start of this function.